src/Rendering/geometryBufferRenderer.ts

```diff
@@ -79,14 +86,103 @@ export class GeometryBufferRenderer {
     private _positionIndex: number = -1;
     private _velocityIndex: number = -1;
     private _reflectivityIndex: number = -1;
+    private _depthNormalIndex: number = -1;
+
+    private _linkedWithPrePass: boolean = false;
+    private _prePassRenderer: PrePassRenderer;
+    private _attachments: number[];
 
     protected _effect: Effect;
     protected _cachedDefines: string;
 
+    /**
+     * @hidden
+     * Sets up internal structures to share outputs with PrePassRenderer
+     * This method should only be called by the PrePassRenderer itself
+     */
+    public linkPrePassRenderer(prePassRenderer: PrePassRenderer) {

```

Comment:
nit: the name of a hidden public method should start with an underscore I think because it is not meant to be used by the end user: @deltakosh can you confirm?